api/prisma/schema.prisma

```diff
@@ -207,19 +207,18 @@ model Topic {
     updatedAt               DateTime           @updatedAt
     parents                 Topic[]            @relation("TopicHierarchy")
     children                Topic[]            @relation("TopicHierarchy")
-    bookmarks               TopicBookmark[]
     publications            Publication[]
 }
 
-model TopicBookmark {
+model Bookmark {
     id            String      @id @default(cuid())
-    topicId       String
+    type          BookmarkType
+    entityId      String
     userId        String
     createdAt     DateTime    @default(now())
-    topic         Topic       @relation(fields: [topicId], references: [id], onDelete: Cascade)
     user          User        @relation(fields: [userId], references: [id], onDelete: Cascade)
 
-    @@unique([topicId, userId])
+    @@unique([entityId, type, userId])
```

Comment:
Not sure we need the type included into the unique constraint. As long as we have the userId and the entityId it should be enough